internal/archive/archive.go

```diff
@@ -20,6 +20,14 @@ type Archive interface {
 	Options() *Options
 	Fetch(pkg string) (io.ReadCloser, error)
 	Exists(pkg string) bool
+	Info(pkg string) (*PackageInfo, error)
+}
+
+type PackageInfo struct {
+	Name    string
+	Version string
+	Arch    string
+	Hash    string
```

Comment:
As a good practice, we typically name such fields after the algorithm used for the hash.

Reply:
Done in https://github.com/canonical/chisel/pull/142/commits/a6628265a33a781995dc24b0dfd5b7731978f65a.